token/program2/src/instruction.rs

```diff
@@ -187,6 +191,26 @@ pub enum TokenInstruction {
     ///   2. `[]` The account's multisignature owner.
     ///   3. ..3+M '[signer]' M signer accounts.
     CloseAccount,
+    /// Freeze an Initialized account or unfreeze a Frozen account, using the Mint's
+    /// freeze_authority (if set).
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   * Single owner
+    ///   0. `[writable]` The account to freeze.
+    ///   1. '[]' The token mint.
+    ///   2. `[signer]` The mint freeze authority.
+    ///
+    ///   * Multisignature owner
+    ///   0. `[writable]` The account to freeze.
+    ///   1. '[]' The token mint.
+    ///   2. `[]` The mint's multisignature freeze authority.
+    ///   3. ..3+M '[signer]' M signer accounts.
+    FreezeAccount {
```

Comment:
Not opposed to `FreezeAccount`/`ThawAccount` without the boolean.  ERC115 uses `Pause`/`Resume`.   I don't have strong opinions here though.  Just surfacing some other ideas

Reply:
2 instructions does feel a little nicer to me, as there's a bit of a mind twist needed to get from Freeze(false) to Thaw. I'll work that up.